openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go

```diff
@@ -0,0 +1,85 @@
+package route
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routevalidation "github.com/openshift/library-go/pkg/route/validation"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const PluginName = "route.openshift.io/ValidateRoute"
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				{Group: routev1.GroupName, Resource: "routes"}: true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				routev1.GroupVersion.WithKind("Route"): routeV1{},
+			})
+	})
+}
+
+func toRoute(uncastObj runtime.Object) (*routev1.Route, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
```

Comment:
nit: i know it's never gong to be `nil` in real-life, do you think we should return an appropriate `field.ErrorList`, or is there a case for `nil`?

Reply:
I see it as harmless, and it's consistent with (i.e., copy-pasted from) comparable functions in a number of other CR validators. Do you think it's worth changing here?

Reply:
no, let's keep it as is.